### PR TITLE
Added functionality to obtain circuit IDs

### PR DESCRIPTION
**Pull request by suhasks123**
### Design:

TOR specifications mention that using *random numbers* to achieve the functionality is the optimal method. This might be for security reasons. But for the scale of our implementation, generating circuit IDs by *incrementing a preset number* is a simpler and more effective way of avoiding circuit ID collisions.

### CODE REVIEW

**Comment by NJnisarg** [comment]
@suhasks123 The master has diverged since you PR. Can you align it before we merge it?

**Comment by suhasks123** [comment]
> @suhasks123 The master has diverged since you PR. Can you align it before we merge it?

I resolved the merge conflicts

File: onion_proxy/main.py [hunk 1]
```diff
@@ -11,14 +11,17 @@
...
 # This function is the actual entry point that will be called
 def main():
 	print("Onion proxy started!")
 	skt = Skt('127.0.0.1', 12346)
 
 	print("Creating a circuit")
-	circ_id = Circuit.get_rand_circ_id()
+	global current_circ_id
+	circ_id = Circuit.get_rand_circ_id(current_circ_id)
+	current_circ_id += 1
```
Comment (arpi-r) [inline onion_proxy/main.py]: Instead of incrementing current_circ_id again, set it to circ_id

**Review by arpi-r — CHANGES_REQUESTED** [review]

File: onion_proxy/main.py (hunk 1 above)
Comment (NJnisarg) [reply onion_proxy/main.py]: @suhasks123 Please use something like a class variable and only one variable. It should not be this hard.

Comment (suhasks123) [reply onion_proxy/main.py]: @NJnisarg So basically remove the get_rand_circ_id() altogether from circuit.py?

Comment (suhasks123) [reply onion_proxy/main.py]: @arpi-r I made the changes

**Pull request merged**